### ISSUE TITLE
Should this work on Linux?

### ISSUE THREAD

**Issue by Asdas91** [OrangeT/vim-csharp#31]
Hi, I wanted to try this plugin on Linux with Mono but I am getting an error. It says "Unable to find csc.exe" on Vim start-up. Is it supposed to work on Linux? (sorry if the question is stupid)

**Comment by jeaye** [comment]
I get this error on OS X Yosemite as well.

**Comment by JamesMcMahon** [comment]
Ditto, same with OS X Yosemite. Any suggestions to fix this issue?

**Comment by 4fthawaiian** [comment]
Same here, MacVim 7-something on Yosemite.

**Comment by rgrinberg** [comment]
Same. Would like to be able to disable this.

**Comment by DennyScott** [comment]
Still have it happening. What exactly is the error from?

**Comment by ramon-rocha** [comment]
I don't build from within Vim but I want syntax highlighting for `cshtml` files.  All I did was suppress the error message by commenting out this line: https://github.com/OrangeT/vim-csharp/blob/master/autoload/cs.vim#L44 with a `"` character.  That only suppresses the error message on startup though.  If you try `:make` it will then indicate that it can't find `msbuild`.

The correct fix is probably to look for `mcs.exe` and `xbuild.exe` respectively on *nix.

**Comment by rbtnn** [comment]
@ramon-rocha Can you build your project on following repository?

https://github.com/rbtnn/vim-csharp

**Comment by HiPhish** [comment]
@rbtnn It works on OS X in Neovim.

**Comment by paralin** [comment]
?

**Comment by jean** [comment]
See https://github.com/marcuskrahl/vim-csharp/tree/mono

**Comment by christianrondeau** [comment]
If that helps anyone, you can add something like this to your `vimrc` to prevent the error. It's a dirty hack though, but works (e.g. for me when doing `vimdiff` in Cygwin)

    let g:net_framework_version = "none" " This is a hack! The string could be whatever you want, we just want to prevent https://github.com/OrangeT/vim-csharp/blob/master/autoload/cs.vim#L44 from happening

**Comment by kianryan** [comment]
Please try latest commits, this should now be resolved.

**Comment by HiPhish** [comment]
I can confirm that it works on OS X 👍

**Comment by kianryan** [comment]
Good to hear.  I'll leave this open for a bit to see if there's anymore feedback.

**Comment by HiPhish** [comment]
Wait, is commit ed34df121e8bf842949fbf880485e4e0994ae602 where all you did was remove
```vim
echom "Unable to find " . a:compiler
```
the solution? You could have at least put in the readme that only Windows is supported.

**Comment by kianryan** [comment]
@HiPhish No.  The solution was multifold.  First it checks for msbuild and xbuild on the path, failing that it looks in known locations on Windows for the latest copy of the framework (since typically they're not on the path by default on Windows).

After that, if it couldn't find msbulid, then it was still being noisy, so it made more sense for it to fail quietly.
